tests/infra-validations.sh

```diff
@@ -200,7 +203,13 @@ git push origin "$infra_generated_version"
 variable_group_exists $AZDO_ORG_URL $AZDO_PROJECT $vg_name "delete"
 
 # Create variable group
-az pipelines variable-group create --name $vg_name --authorize true --variables "ACCESS_TOKEN_SECRET=$ACCESS_TOKEN_SECRET" "ARM_CLIENT_ID=$SP_APP_ID" "ARM_CLIENT_SECRET=$SP_PASS" "ARM_TENANT_ID=$SP_TENANT" "CLUSTER=$infra_region" "GENERATED_REPO=https://$repo_url" "PROJECT_DIRECTORY=$infra_hld_project" "AZDO_ORG_NAME=$AZDO_ORG_URL" "AZDO_PROJECT_NAME=$AZDO_PROJECT" "ARM_SUBSCRIPTION_ID=$SP_SUBSCRIPTION_ID"
+variable_group_id=$(az pipelines variable-group create --name $vg_name --authorize true --variables  "ARM_TENANT_ID=$SP_TENANT" "CLUSTER=$infra_region" "GENERATED_REPO=https://$repo_url" "PROJECT_DIRECTORY=$infra_hld_project" "AZDO_ORG_NAME=$AZDO_ORG_URL" "AZDO_PROJECT_NAME=$AZDO_PROJECT" | jq '.id')
+
+# Update secret variables in variable group
+variable_group_variable_create $variable_group_id $AZDO_ORG_URL $AZDO_PROJECT "ACCESS_TOKEN_SECRET" $ACCESS_TOKEN_SECRET "secret"
+variable_group_variable_create $variable_group_id $AZDO_ORG_URL $AZDO_PROJECT "ARM_CLIENT_ID" $SP_APP_ID "secret"
+variable_group_variable_create $variable_group_id $AZDO_ORG_URL $AZDO_PROJECT "ARM_CLIENT_SECRET" $SP_PASS "secret"
+variable_group_variable_create $variable_group_id $AZDO_ORG_URL $AZDO_PROJECT "ARM_SUBSCRIPTION_ID" $SP_SUBSCRIPTION_ID "secret"
```

Comment:
[question] do we need to have `$SP_SUBSCRIPTION_ID` as secret. I looked at other place where  `AZ_STORAGE_ACCOUNT` is not a secret. IMO, we need to be consistent here.

Reply:
I don't believe `$SP_SUBSCRIPTION_ID` is used anywhere else since this is a requirement that we pass to the generate pipeline to run terraform commands. I figured it would be best to have either Tenant ID or Subscription ID as a secret but happy to remove. Thoughts?

Reply:
IMO, we can remove it. @andrebriggs ?

Reply:
Yeah let's just remove.

Reply:
Removed